guava/src/com/google/common/hash/AbstractHashFunction.java

```diff
@@ -54,24 +61,27 @@ public HashCode hashLong(long input) {
     return newHasher(8).putLong(input).hash();
   }
 
+  @SuppressWarnings("upperbound:argument.type.incompatible")// If input has min length of 1, since off is 0, off + input.length - 1 is always < input.length.
   @Override
-  public HashCode hashBytes(byte[] input) {
+  public HashCode hashBytes(byte @MinLen(1)[] input) {
     return hashBytes(input, 0, input.length);
   }
 
   @Override
-  public HashCode hashBytes(byte[] input, int off, int len) {
+  public HashCode hashBytes(byte[] input, @NonNegative @LTLengthOf(value = "#1", offset = "#3 - 1") int off, @NonNegative @LTLengthOf(value = "#1", offset = "#2 - 1") int len) {
     checkPositionIndexes(off, off + len, input.length);
     return newHasher(len).putBytes(input, off, len).hash();
   }
 
   @Override
+  @SuppressWarnings("lowerbound:argument.type.incompatible")//ByteBuffer#remaining is annotated to return non negative value
+  //Link to PR: https://github.com/typetools/checker-framework/pull/2080 ( status: waiting for merge)
```

Comment:
This PR has been merged, so we should be able to remove this warning suppression.